### PR TITLE
Improve the search and search components of the Feed screen.

### LINKED ISSUE #36: Feed screen Search component
Improve the search and search components of the Feed screen.

Screen: Feed
Component: SearchComponent

- [] Use the SearchComponent to update the kinds [NDKKind] that are being used in the Feed useNote hooks.
- [] Open menu
- [] Buttons below the search to add/remove kind of: NDKKind.Text, NDKKind.ChannelCreation, NDKKind.GroupChat, NDKKind.ChannelMessage, NDKKind.Metadata

Images:

image

### PR DESCRIPTION
**Pull request by Dprof-in-tech**
closes #36 

This is what it looks like right now
image

### CODE REVIEW

File: apps/mobile/src/components/search/index.tsx [hunk 1]
```diff
@@ -41,11 +72,33 @@ const SearchComponent = ({searchQuery, setSearchQuery}: ISearchComponent) => {
...
+        onClose={() => setIsOpenFilter(false)}
+        kinds={kinds}
+        setKinds={setKinds}
+        onSortChange={handleSortChange}
+        activeSortBy={activeSortBy}
+      />
+
+      {isLoading && <Text>Loading...</Text>}
+      {isError && <Text>Error loading data</Text>}
+      {data?.pages?.map((page, index) => (
```
Comment (MSghais) [inline apps/mobile/src/components/search/index.tsx:95]: We have already a Flat list, you need to add the filter and search on the Feed not here

File: apps/mobile/src/screens/Feed/index.tsx [hunk 2]
```diff
@@ -17,73 +17,45 @@ export const Feed: React.FC<FeedScreenProps> = ({navigation}) => {
...
-      </View> */}
-
-      <SearchComponent setSearchQuery={setSearch} searchQuery={search}></SearchComponent>
-
-      {/* Todo todo filter for trending, latest etc */}
+      <SearchComponent
+        setSearchQuery={setSearch}
+        searchQuery={search ?? ''}
+        kinds={kinds}
+        setKinds={setKinds}
```
Comment (MSghais) [inline apps/mobile/src/screens/Feed/index.tsx:50]: add contact list here

File: apps/mobile/src/screens/Feed/index.tsx [hunk 3]
```diff
@@ -92,32 +64,18 @@ export const Feed: React.FC<FeedScreenProps> = ({navigation}) => {
...
-                  // image={item.img}
-                  event={item}
-                />
-              );
-            }}
+            renderItem={({item}) => <BubbleUser event={item} />}
           />
         }
         contentContainerStyle={styles.flatListContent}
         data={notes.data?.pages.flat()}
```
Comment (MSghais) [inline apps/mobile/src/screens/Feed/index.tsx:72]: add new notes with search here

**Review by MSghais — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/Feed/index.tsx (hunk 3 above)
Comment (Dprof-in-tech) [reply apps/mobile/src/screens/Feed/index.tsx:72]: done ser

File: apps/mobile/src/screens/Feed/index.tsx (hunk 2 above)
Comment (Dprof-in-tech) [reply apps/mobile/src/screens/Feed/index.tsx:50]: done ser

File: apps/mobile/src/screens/Feed/index.tsx [hunk 4]
```diff
@@ -17,115 +17,77 @@ export const Feed: React.FC<FeedScreenProps> = ({navigation}) => {
...
+        setSearchQuery={setSearch}
+        searchQuery={search ?? ''}
+        kinds={kinds}
+        setKinds={setKinds}
+        contactList={profilesSearch.map((item) => item?.id)}
+      />
 
       <FlatList
-        ListHeaderComponent={
-          <FlatList
```
Comment (MSghais) [inline apps/mobile/src/screens/Feed/index.tsx:83]: Why you delete the user view in story mode?

File: apps/mobile/src/screens/Feed/index.tsx [hunk 5]
```diff
@@ -17,115 +17,77 @@ export const Feed: React.FC<FeedScreenProps> = ({navigation}) => {
...
-          />
-        }
         contentContainerStyle={styles.flatListContent}
-        data={notes.data?.pages.flat()}
+        data={combinedData}
         keyExtractor={(item) => item?.id}
         renderItem={({item}) => {
-          if (!item?.content?.includes(search) && search && search?.length > 0) return <></>;
-          if (item?.kind == NDKKind.ChannelCreation || item?.kind == NDKKind.ChannelMetadata) {
+          if (item.kind === undefined) {
```
Comment (MSghais) [inline apps/mobile/src/screens/Feed/index.tsx:66]: kind === undefined?

Comment (MSghais) [reply apps/mobile/src/screens/Feed/index.tsx:66]: Yes not render the BubbleUser here please..

**Review by MSghais — CHANGES_REQUESTED** [review]

**Comment by Dprof-in-tech** [comment]
i just put back the original structure for the feed component

**Comment by MSghais** [comment]
Well the PR is not finish yet. 
We are gonna merge as it is and finish at after, thank you @Dprof-in-tech

**Review by MSghais — APPROVED** [review]

**Pull request merged**